jenkins/opensearch/integ-test.jenkinsfile

```diff
@@ -0,0 +1,133 @@
+lib = library(identifier: "jenkins@20211118", retriever: legacySCM(scm))
+
+pipeline {
+    agent none
+    environment {
+        AWS_ROLE_ARN = "arn:aws:iam::${AWS_ACCOUNT_PUBLIC}:role/opensearch-test"
+        AWS_ROLE_SESSION_NAME = "jenkins-test-session"
+    }
+    tools {
+        jdk "JDK14"
+        maven "maven-3.8.2"
+    }
+    parameters {
+        string(
```

Comment:
Aren't many of these redundant with `ARTIFACT_BASE_PATH`? I feel like the only thing I want to supply to an integ test build is the location of the build (a manifest).

Reply:
Agreed, working on simplifying these:
`OPENSEARCH_VERSION`: can be removed
`ARCHITECTURE`: used for selecting the env to run the tests in (could be generated from `ARTIFACT_BASE_PATH`)
`BUILD_ID`: used in `uploadTestResults`
`TEST_MANIFEST`: needed for running the actual `test.sh` script
`BUILD_MANIFEST`: used in `uploadTestResults` (could be generated from `ARTIFACT_BASE_PATH`)

From my understanding, it can be better to pass the full strings than dynamically generate them, to help prevent broken links in the future, should any of the artifact path naming conventions change, etc. But if these conventions are unlikely to change then I'm ok to generate them directly from the base path too, not sure on the status of that.

Lmk what you think. cc @peternied

Reply:
BUILD_ID is _what_ is being tested? In that case the build manifest contains it, no? 

I think if you give a build manifest and a test manifest that makes sense.

Reply:
Right, BUILD_ID can be pulled from the build manifest too. I'm ok to simplify to just build manifest and test manifest. I'll add some helper fns to pull out the rest of the info.